### ISSUE TITLE
ContentType Middleware not being detected from CRD

### ISSUE THREAD

**Issue by Carles-Figuerola** [containous/traefik#7005]
### Do you want to request a *feature* or report a *bug*?

Bug

The Middleware of type "ContentType" does not seem to be loaded properly when defined using a CRD. 

The TYPE field on the dashboard shows blank and when using them from an IngressRoute, the dashboard says that it can't be found/

### What did you do?

I can copy over the content, but there is an explanation with details on the community forums:
https://community.containo.us/t/traefik-and-grpc-sending-to-the-wrong-port-using-https-h2c/4656/3

### Output of `traefik version`: (_What version of Traefik are you using?_)
 
versions 2.1 and 2.2 seems to have this issue

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Using kubernetes 1.14 with CRD provider

**Comment by johnpekcan** [comment]
Thank you for submitting this @Carles-Figuerola

**Comment by johnpekcan** [comment]
Okay, I believe we need to add `ContentType:       middleware.Spec.ContentType` in [here](https://github.com/containous/traefik/blob/master/pkg/provider/kubernetes/crd/kubernetes.go#L212-L233).

Doing so, seems to let me load the middleware in without a fuss:

![image](https://user-images.githubusercontent.com/8002714/87256278-bfa95a80-c456-11ea-867b-2cde10ade680.png)

I was able to call it in from an entrypoint as well:
![image](https://user-images.githubusercontent.com/8002714/87256305-eebfcc00-c456-11ea-897f-ad71b6b271a4.png)


Though I'm not entirely sure it's behaving correctly, or if this is user error. I'm still seeing:
```console
➜  ~ (⎈ |minikube:argocd) argocd login --insecure  localhost:8000
Username: admin
Password:
FATA[0005] rpc error: code = Internal desc = transport: received the unexpected content-type "text/plain; charset=utf-8"
```

**Comment by johnpekcan** [comment]
Tagged a pr for this issue, want to see if this alleviates your issue @Carles-Figuerola ?
It's entirely possible mine is on my argocd setup 😅

**Comment by Carles-Figuerola** [comment]
Hi @johnpekcan, I can test that change with my setup. However, I do not currently have the dev environment to build the tagged PR, so if you could share the amd64 binary that would definitely speed me up a lot :)

**Comment by johnpekcan** [comment]
@Carles-Figuerola  
Where are my manners, please give this a go! 
This is shared publicly from my google drive -- please let me know if this violates any terms or anything 😅 
https://drive.google.com/file/d/19lY14QdWr2AD3hpLEN0NHI-QonoC3Cqq/view?usp=sharing

**Comment by traefiker** [comment]
Closed by #7034.
